### PR TITLE
[4.0] convert to prepared statements UserGroup table

### PR DESCRIPTION
**Pull request by alikon**
### Summary of Changes
converted sql to prepared statements

### Testing Instructions
-  create user group
-  delete user group
-  modify user group
-  view user group

### Expected result
works as before

### Actual result
N/A

### CODE REVIEW

File: libraries/src/Table/Usergroup.php [hunk 1]
```diff
@@ -117,9 +133,20 @@ public function rebuild($parent_id = 0, $left = 0)
...
+		$right = (int) $right;
+
 		// We've got the left value, and now that we've processed
 		// the children of this node we also know the right value
-		$db->setQuery('UPDATE ' . $this->_tbl . ' SET lft=' . (int) $left . ', rgt=' . (int) $right . ' WHERE id=' . (int) $parent_id);
+		$query->clear()
+			->update($db->quoteName($this->_tbl))
+			->set($db->quoteName('lft') . ' = :lft')
+			->set($db->quoteName('rgt') . ' = :rgt')
+			->where($db->quoteName('id') . '= :id')
```
Comment (Quy) [inline libraries/src/Table/Usergroup.php]:
```suggestion
			->where($db->quoteName('id') . ' = :id')
```

**Comment by Quy** [comment]
Create a new group with Group Parent `Administrator`. It is not listed under `Administrator`.

**Comment by chmst** [comment]
Same effect as @Quy <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28491">issues.joomla.org/tracker/joomla-cms/28491</a>.</sub>

**Comment by alikon** [comment]
fixed

**Comment by chmst** [comment]
In a fresh installation, I deleted 2 usergroups at the same time and got this: 
delete-usergroups

**Comment by Quy** [comment]
@chmst Please explain why your last report is an issue.

**Comment by chmst** [comment]
There is no menu any more, no toggle icon and no modules in the header.

**Comment by Quy** [comment]
It is not happening to me. I will install the latest version and test again.

**Comment by chmst** [comment]
Probably this is caused by other PRs in my system. But I had this effect several times. I'll re-test asap with a clean installation

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 28c9b9ee3564355268fa5ee8ce4c6a6f7e4264ef<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28491">issues.joomla.org/tracker/joomla-cms/28491</a>.</sub>

**Comment by chmst** [comment]
Sorry, but I got this again on a fresh installation. I deleted groups with child groups: manager and editor. The result is as in my screen above. But it is nor related to this PR, It is also before theis PR.
Will make a new issue for it.

**Comment by chmst** [comment]
I have tested this item :white_check_mark: successfully on 28c9b9ee3564355268fa5ee8ce4c6a6f7e4264ef<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28491">issues.joomla.org/tracker/joomla-cms/28491</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28491">issues.joomla.org/tracker/joomla-cms/28491</a>.</sub>

**Pull request merged**